source/detail/serialization/xlsx_consumer.cpp

```diff
@@ -298,7 +307,11 @@ std::pair<xlnt::row_properties, int> parse_row(xml::parser *parser, xlnt::detail
         }
         else if (string_equal(attr.first.name(), "s"))
         {
-            props.first.style = strtoul(attr.second.value.c_str(), nullptr, 10);
+            unsigned long long style = 0;
+            if (xlnt::detail::parse(attr.second.value, style))
+            {
+                props.first.style = static_cast<size_t>(style);
```

Comment:
Here we could simplify the code as you proposed in a few other cases, meaning that:
- if `size_t` is used on a 32-bit system, it should map to `unsigned int`
- if `size_t` is used on a 64-bit system that has 64-bit `unsigned long`, it will probably use `unsigned long` (Linux, macOS)
- if `size_t` is used on a 64-bit system that has 32-bit `unsigned long`, it will probably use `unsigned long long` (Windows)

Since we have parsers for all types, it shouldn't be an issue. I can then simplify the code to directly parse the `size_t`, just like you proposed in a few other cases.

Note: actually, I just looked it up and the `"s"` attribute of `"row"` is an `unsignedInt`, which is a [32-bit unsigned integer in XML](http://www.datypic.com/sc/xsd/t-xsd_unsignedInt.html). In other words, we could think about whether we should switch from `size_t` to `unsigned int` in this case.

Reply:
> Note: actually, I just looked it up and the "s" attribute of "row" is an unsignedInt, which is a [32-bit unsigned integer in XML](http://www.datypic.com/sc/xsd/t-xsd_unsignedInt.html). In other words, we could think about whether we should switch from size_t to unsigned int in this case.

As long as the public interface doesn't change, I'm fine with both, but if we change the type here, there are probably other places that should be changed too if we want to be consistent.

Reply:
Unfortunately, `xlnt::row_properties` is a public class and its member variables are publicly accessible. While changing from `xlnt::optional<size_t>` to `xlnt::optional<unsigned int>` would mean that in most cases no changes should be necessary, this change might be problematic for existing external code (outside of XLNT) that might use templates or casts. While we should be able to easily change it in the future, I think that it does not make sense yet for this PR.

I'll just do my other proposed change and take advantage of the fact that `size_t` will map to either `unsigned int`, `unsigned long` or `unsigned long long`, types for which we already have parsers.